lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -333,16 +333,24 @@ void SessionState::sess_infocopy(struct SessionInfo* info) {
   // Static SessionInfo vlaue till UPF node value implementation
   // gets stablized.
   // TODO we cud eventually  migrate to SMF-UPF proto enum directly.
-  info->state = get_proto_fsm_state();
-  info->subscriber_id.assign(imsi_);
-  info->ver_no              = get_current_version();
-  info->local_f_teid        = get_upf_local_teid();
-  info->nodeId.node_id_type = SessionInfo::IPv4;
-  info->pdr_rules           = get_all_pdr_rules();
-  if (!info->pdr_rules.empty()) {
-    // Get the UE ip address from first rule
-    auto& rule    = info->pdr_rules.front();
-    info->ip_addr = rule.pdi().ue_ipv4();
+  if (info != nullptr) {
+    info->state = get_proto_fsm_state();
+    info->subscriber_id.assign(imsi_);
+    info->ver_no       = get_current_version();
+    info->local_f_teid = get_upf_local_teid();
+    info->pdr_rules    = get_all_pdr_rules();
+    if (!info->pdr_rules.empty()) {
+      // Get the UE ip address from first rule
+      auto& rule = info->pdr_rules.front();
+      if (SessionInfo::IPv4) {
```

Comment:
What does this line check for?

Reply:
How are these values set?

Reply:
Should have checked for `!rule.pdi().ue_ipv4().empty()` from the start, mistakenly added `SessionInfo::IPv4`. Fixed it now.